src/braket/analog_hamiltonian_simulator/rydberg/validators/program.py

```diff
@@ -34,7 +39,64 @@ def local_detuning_pattern_has_the_same_length_as_atom_array_sites(cls, values):
             pattern_size = len(local_detuning["magnitude"]["pattern"])
             if num_sites != pattern_size:
                 raise ValueError(
-                    f"The length of pattern ({pattern_size}) of shifting field {idx} must equal "
+                    f"The length of pattern ({pattern_size}) of local detuning {idx} must equal "
                     f"the number of atom array sites ({num_sites})."
                 )
         return values
+
+    # If there is local detuning, the net value of detuning for each atom
+    # should not exceed a certain value
+    @root_validator(pre=True, skip_on_failure=True)
+    def net_detuning_must_not_exceed_max_net_detuning(cls, values):
```

Comment:
Should we split this function? Also, let's discuss more on this offline.

Reply:
Few questions we discussed offline and we might need confirmation from the science team on these: 
1. Where to cross-verify those schema values, trying to understand what does these detuning patterns, magnitude mean?